server/src/main/java/org/apache/iotdb/db/qp/executor/PlanExecutor.java

```diff
@@ -914,8 +914,26 @@ public void insert(InsertPlan insertPlan) throws QueryProcessException {
       insertPlan.setSchemasAndTransferType(schemas);
       StorageEngine.getInstance().insert(insertPlan);
       if (insertPlan.getFailedMeasurements() != null) {
-        throw new StorageEngineException(
-            "failed to insert points " + insertPlan.getFailedMeasurements());
+        // check if all path not exist exceptions
+        List<String> failedPaths = new ArrayList<>(insertPlan.getFailedMeasurements().keySet());
+        List<Exception> exceptions = new ArrayList<>(insertPlan.getFailedMeasurements().values());
+        boolean isPathNotExistException = true;
+        for(Exception e : exceptions){
+          Exception curException = e;
+          while(curException.getCause() != null){
+            curException = (Exception) curException.getCause();
+          }
+          if(!(curException instanceof PathNotExistException)){
+            isPathNotExistException = false;
+            break;
+          }
+        }
+        if(isPathNotExistException){
+          throw new PathNotExistException(failedPaths.toString());
```

Comment:
I suggest you just add a field of List<String> in PathNotExistException, since concatenating a long string list is time-consuming.

Besides, please perform a reformat.

Reply:
An attribute is added in the PathNotExistExeption. Thanks for your suggestion.